src/envoy/utils/grpc_transport.cc

```diff
@@ -43,6 +45,22 @@ GrpcTransport<RequestType, ResponseType>::GrpcTransport(
             request.DebugString());
 }
 
+template <class RequestType, class ResponseType>
+void GrpcTransport<RequestType, ResponseType>::onCreateInitialMetadata(
+    Http::HeaderMap &metadata) {
+  // We generate cluster name contains invalid characters, so override the
+  // authority header temorarily until it can be specified via CDS.
+  // See https://github.com/envoyproxy/envoy/issues/3297 for details.
+  metadata.Host()->value("mixer", 5);
+
+  if (!forward_attributes_.attributes().empty()) {
+    HeaderUpdate header_update_(&metadata);
+    std::string serialized_attributes;
+    forward_attributes_.SerializeToString(&serialized_attributes);
```

Comment:
maybe we can optimize this?   only do searizliedToString at init time,  not on every request.

Reply:
Tried. My testing set-up requires pushing docker images and my C++ skills lead to some segfaults for the optimized version. Let's do optimization in another PR when I have a simpler test set-up.